File: hw/sivinta/U3/HW22Observer/ObserverPython/src/view/application.py
"""
Main application class to demonstrate the Observer pattern
@author Gerald Astudillo
"""

import sys
import os

# Add the src directory to the Python path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))

from model.ibm import IBM
from model.investor import Investor

class Application:
    @staticmethod
    def main():
        # Create investors (observers)
        s = Investor("Sorros")
        b = Investor("Berkshire")

        # Create IBM stock (subject)
        ibm = IBM("IBM", 120.00)
        ibm.add_observer(s)
        ibm.add_observer(b)

        # Change prices and notify observers
        ibm.set_price(120.10)
        ibm.set_price(121.00)
        ibm.set_price(120.50)
        ibm.set_price(120.75)
        ibm.set_symbol("IBMTEST")

if __name__ == "__main__":
    Application.main()

File: hw/andrade/u3/HW23Observer/HW23Observer/src/main/java/ec/edu/espe/hw23observer/view/Application.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package ec.edu.espe.hw23observer.view;

import ec.edu.espe.hw23observer.model.IBM;
import ec.edu.espe.hw23observer.model.Investor;

/**
 *
 * @author Julio Andrade
 */
public class Application {

    public static void main(String[] args) {
        Investor s = new Investor("Sorros");
        Investor b = new Investor("Berkshire");

        IBM ibm = new IBM("IBM", 140.00);
        ibm.addObserver(s);
        ibm.addObserver(b);

        ibm.setPrice(140.10);
        ibm.setPrice(141.00);
        ibm.setPrice(140.70);
        ibm.setPrice(140.65);
        ibm.setSymbol("IBMTEST");
    }
}
